avocado/core/test.py

```diff
@@ -92,26 +96,39 @@ def __eq__(self, other):
         else:
             return self.__dict__ == other.__dict__
 
+    @property
     def str_filesystem(self):
         """
-        File-system friendly representation of the test name
+        Test ID in a format suitable for use in file systems
+
+        The string returned should be safe to be used as a file or
+        directory name.  This file system version of the test ID may
+        have to shorten either the Test Name or the Variant ID.
+
+        The first component of a Test ID, the numeric unique test id,
+        AKA "uid", will be used as a an stable identifier between the
+        Test ID and the file or directory created based on the return
+        value of this method.  If the filesystem can not even
+        represent the "uid", than an exception will be raised.
+
+        For Test ID "001-mytest;foo", examples of shortened file
+        system versions include "001-mytest;f" or "001-myte;foo".
```

Comment:
I noticed that this is actually incorrect.  When a Test ID contains a variant, the shortening will happen initially on the test name *only*. Only after it is exhausted (test name is removed), then the variant id will be shortened.